adaway-linux.sh

```diff
@@ -39,53 +39,59 @@ fi
 
 # preparing temporary directory
 if [ -d "${TMPDIR}" ]; then
-    echo "[i] deleting directory ${TMPDIR}"
+    echo "[i] Deleting directory ${TMPDIR}"
     rm -r "${TMPDIR}"
 fi
 
-echo "[i] creating temporary directory ${TMPDIR}"
+echo "[i] Creating temporary directory ${TMPDIR}"
 mkdir -p "${TMPDIR}"
 
 # add domains from hosts-server listet in hostssources.lst
 while read src; do
     if [[ $src != "#*" ]] ; then
         # only insert entries redirecting to 127.0.0.1 or 0.0.0.0 (everything else might be a security nightmare)
-        curl --progress-bar -L "${src}" | sed 's/\r/\n/' | sed 's/\s\+/\t/' | sed 's/0\.0\.0\.0/127.0.0.1/' | grep '127\.0\.0\.1' >> "${TMPDIR}/hosts.downloaded"
+        curl --progress-bar -L "${src}" | sed 's/\r/\n/' | sed 's/\s\+/\t/' | sed 's/0\.0\.0\.0/127.0.0.1/' | grep '127\.0\.0\.1' >> "${TMPDIR}hosts.downloaded"
     else
         echo "[i] skipping $src"
     fi
 done <hostssources.lst
 
+echo "[i] Cleanup ${TMPDIR}hosts.downloaded and merge with original content"
+grep '^\([0-9]\|:\)' "${TMPDIR}hosts.downloaded" > "${TMPDIR}temp" && mv "${TMPDIR}temp" "${TMPDIR}hosts.downloaded" # Remove all lines that are not hosts entries
```

Comment:
I guess the `:` is for IPv6 Addresses? I think it's quite hard to catch all IP addresses that will work (e.g. there could be `::0` or `0:0:0::` or ...). We could rewrite everything in python and do something like:

```python
import ipaddress
addr = ip_address(input_addr)
if addr.is_unspecified or addr.is_loopback:
    ...
```